src/roadmapper/milestone.py

```diff
@@ -59,6 +58,38 @@ def draw(self, painter: Painter) -> None:
                 self.diamond_height,
                 self.fill_colour,
             )
+
+        alignment = (
+            self.text_alignment.lower() if self.text_alignment is not None else None
+        )
+
+        if alignment is None or alignment == "centre":
+            pass  # Text is already "centre" if we change nothing
+        elif "left" in alignment or "right" in alignment:  # Handle left/right
+            text_width, _ = painter.get_text_dimension(
+                text=self.text, font=self.font, font_size=self.font_size
+            )
+
+            if ":" in alignment:  # Split if ":" in str
+                alignment, modifier = alignment.split(":")
+
+                if "%" in modifier:  # If "%" treat as percent of text_width
+                    offset = (int(modifier.strip("%")) / 100) * text_width
+                else:  # Treat as units
+                    offset = int(modifier)
+            else:  # If no ":", default to half of text width
+                offset = 0.5 * text_width
```

Comment:
Considering the ideal of clean code, we do not really want nested ifs. They are just more complex to understand and can cause failures when refactoring code. If you wanted to stick to the current solution, I would simply create submethods for this block. This makes it easier to know what this block does without having to think about the details. Additionally, method names are like the better comments: you have description text which has to be refactored if you change something, unlike comments which tend to be forgotten.

Without forcing you to stick to my suggestion, I could think of something like the following 
```
def handle_correct_cases_of(alignment):
    if alignment is None or alignment == "centre":
        return  # Text is already "centre" if we change nothing

    offset = get_offset_from(alignment)
    apply_offset_for(alignment, offset)


def get_offset_from(alignment):
    # Calculat correct offset value
    # Keep in mind that a `return` can eliminate the need for an else/elif
    # if something is x:
    #     return x
    # no_need_for_else()
    # return 5


def apply_offset_for(alignment, offset):
    # Apply offset to self.text_x
```

Reply:
This is good feedback and considering that either the parsing of more complex alignments makes sense to add in multiple places, or it makes no sense to add it, I have refactored Alignment it out to it's own class.